src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -145,23 +151,36 @@ public async Task<ITriggerData> BindAsync(object value, ValueBindingContext cont
                     valueProvider = new HttpRequestValueBinder(_parameter, request, invokeString);
                 }
 
-                var bindingData = await GetBindingDataAsync(request, valueProvider);
+                object poco = null;
```

Comment:
To make the binding data lazy, I have to hoist the async evaluation portions out, since the TriggerData.BindingData interface isn't async.